### ISSUE TITLE
Add dolly-v2 models

### ISSUE THREAD

**Issue by wxl-lxw** [Lightning-AI/lit-gpt#692]
Excuse me, could you please add dolly-v2 models, since I really want to use PEFT to fine-tune dolly-v2

Thank you so much!

**Comment by aniketmaurya** [comment]
@carmocca I guess it would be helpful to have a dynamic weight loader for the supported models. dolly-v2 is trained on Pythia which is supported by Lit-GPT.

**Comment by Andrei-Aksionov** [comment]
Yeah, I also think that it might be helpful to have something like
```bash
python scripts/download.py --repo_id databricks/dolly-v2-3b --extends-config EleutherAI/pythia-2.8b
```
so we don't have to add configs for all the possible “descendants”.
But perhaps for now I think it's better to stick to the existing approach.

@carmocca Maybe restore #427 and put it there? Plus `StableBeluga2` config hasn't been added but the issue was close for some reason 🤷‍♂️.

**Comment by carmocca** [comment]
I prefer to do the simple thing of having dicts per supported config instead of trying to be smart and making an automated system for it. We can always refactor later if we choose to.

It doesn't seem like StableBeluga2 was popular, so I didn't mind closing the issue. 

@wxl-lxw Feel free to add support for this model. Looks like adding a config for it should be enough

**Comment by Andrei-Aksionov** [comment]
In addition want to mention that in #427 one can find links with an example of how to do a similar PR. 
It's a fun an easy way to start contributing.

@wxl-lxw feel free to create a PR and, if needed, someone will help you along the way.

**Comment by mehrdad-es** [comment]
@carmocca @Andrei-Aksionov , how do you determine if just copying the appropriate config would be enough to support the model on lit-gpt vs models that would need additional modifications and cannot be currently supported?

**Comment by carmocca** [comment]
I simply look at the reference implementation to see if the code is equivalent

**Comment by mehrdad-es** [comment]
@carmocca  I think in another issue you mentioned models using dropouts are not currently supported in lit-gpt. I was confused because most transformer models have dropout in them and lit-gpt doesn't have it? 

Sincerely,
Mehrdad

**Comment by carmocca** [comment]
Dropout is only relevant during training, and yes, we don't support it.

So far the only model that we support that uses it is Phi-2 (#885). However, some of the released weights do not provide training details so it is possible that they were also using dropout.

I wouldn't say that the current trend of models uses it, especially the larger models that are trained on trillions of tokens.
